src/wrapper/api_helpers.jl

```diff
@@ -1,59 +1,64 @@
-# 
-# function Highs_getHighsBoolOptionValue(highs, option, value)
-    # ccall((:Highs_getHighsBoolOptionValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cint}), highs, option, value)
-# end
-# 
-# function Highs_getHighsIntOptionValue(highs, option, value)
-    # ccall((:Highs_getHighsIntOptionValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cint}), highs, option, value)
-# end
-# 
-# function Highs_getHighsDoubleOptionValue(highs, option, value)
-    # ccall((:Highs_getHighsDoubleOptionValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cdouble}), highs, option, value)
-# end
-# 
-# function Highs_getHighsStringOptionValue(highs, option, value)
-    # ccall((:Highs_getHighsStringOptionValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Cstring), highs, option, value)
-# end
-# 
-# function Highs_getHighsIntInfoValue(highs, info, value)
-    # ccall((:Highs_getHighsIntInfoValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cint}), highs, info, value)
-# end
-# 
-# function Highs_getHighsDoubleInfoValue(highs, info, value)
-    # ccall((:Highs_getHighsDoubleInfoValue, libhighs), Cint, (Ptr{Cvoid}, Cstring, Ptr{Cdouble}), highs, info, value)
-# end
-# 
-# function Highs_getSolution(highs, colvalue, coldual, rowvalue, rowdual)
-    # ccall((:Highs_getSolution, libhighs), Cvoid, (Ptr{Cvoid}, Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Cdouble}), highs, colvalue, coldual, rowvalue, rowdual)
-# end
-# 
-# function Highs_getBasis(highs, colstatus, rowstatus)
-    # ccall((:Highs_getBasis, libhighs), Cvoid, (Ptr{Cvoid}, Ptr{Cint}, Ptr{Cint}), highs, colstatus, rowstatus)
-# end
 
+"""
+    set_option(highs, option, value::T)
+
+Sets the option to the value, dispatches on
+the value type T to call appropriate HiGHS function.
+"""
+function set_option end
+
+function set_option(highs, option, value::Integer)
+    Highs_setHighsIntOptionValue(highs, option, Cint(value))
+end
+
+function set_option(highs, option, value::Bool)
+    Highs_setHighsBoolOptionValue(highs, option, Cint(value))
+end
+
+function set_option(highs, option, value::AbstractFloat)
+    Highs_setHighsDoubleOptionValue(highs, option, Cdouble(value))
+end
+
+function set_option(highs, option, value::String)
+    Highs_setHighsStringOptionValue(highs, option, value)
+end
+
+"""
+    get_option(highs, option, T)
+
+Gets the option of type T on the HiGHS side.
+"""
+function get_option end
+
+function get_option(highs, option, ::Type{T}) where {T <: Integer}
+    value = Ref{Cint}(0)
+    Highs_getHighsIntOptionValue(highs, option, value)
+    return T(value[])
+end
+
+function get_option(highs, option, ::Type{Bool})
+    value = Ref{Cint}(0)
+    Highs_getHighsBoolOptionValue(highs, option, value)
+    return Bool(value[])
+end
+
+function get_option(highs, option, ::Type{T}) where {T <: AbstractFloat}
+    value = Ref{Cdouble}()
+    Highs_getHighsDoubleOptionValue(highs, option, value)
+    return T(value[])
+end
+
+
+function get_option(highs, option, ::Type{String})
+    p = pointer("")
```

Comment:
This code could cause segfaults. The buffer passed to getHighsStringOptionValue needs to be sufficiently large to hold the result. The API doesn't tell you how long this needs to be, but we could use 100 characters to be relatively safe.

Reply:
ok just fixed it. Given that the buffer passed was of size 0 (1 with the null termination), I assumed it got resized / that String had a safety buffer anyway